app/src/main/java/comp3350/mealbuddy/presentation/TimelineActivity.java

```diff
@@ -1,17 +1,117 @@
 package comp3350.mealbuddy.presentation;
 
-import comp3350.mealbuddy.R;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.cardview.widget.CardView;
 
-import android.os.Bundle;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
+import java.time.LocalDate;
+import java.util.Calendar;
+import java.util.Date;
+
+import comp3350.mealbuddy.R;
+import comp3350.mealbuddy.business.AccessAccount;
+import comp3350.mealbuddy.business.Calculator;
+import comp3350.mealbuddy.objects.Day;
 
 
 public class TimelineActivity extends AppCompatActivity {
 
+    private AccessAccount accessAccount;
+    private Day day;
+    private Date today;
+    private LocalDate localDate;
+    private Calculator calculator;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
         setContentView(R.layout.activity_timeline);
+        final String username = this.getIntent().getStringExtra("username");
+        Calendar calendar = Calendar.getInstance();
+        accessAccount = new AccessAccount();
+        day = accessAccount.getDay(accessAccount.getAccount(username), calendar.get(Calendar.DAY_OF_YEAR));
+        System.err.println(day.dayOfYear);
+        calculator = new Calculator(day);
+        initializeCards();
+        FloatingActionButton addFood = (FloatingActionButton) findViewById(R.id.btnAddFood);
+        addFood.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(TimelineActivity.this, AddFoodActivity.class);
+                intent.putExtra("dayOfYear", day.dayOfYear);
+                intent.putExtra("username", username);
+                TimelineActivity.this.startActivity(intent);
+            }
+        });
+    }
```

Comment:
you can change this to a lambda function.. (please test)
```suggestion
        addFood.setOnClickListener((view) -> {
                Intent intent = new Intent(TimelineActivity.this, AddFoodActivity.class);
                intent.putExtra("dayOfYear", day.dayOfYear);
                intent.putExtra("username", username);
                TimelineActivity.this.startActivity(intent);
            });
    }
```

Reply:
Ignore the above: I removed the suggestion, this can only be done in Java8 and we are on Java7. We must confirm with Braico if we can go to 8